runhouse/resources/hardware/launcher_utils.py

```diff
@@ -239,17 +242,46 @@ def up(cls, cluster, verbose: bool = True, force: bool = False):
         data = read_resp_data(resp)
         logger.info("Successfully launched cluster")
         cls._update_from_den_response(cluster=cluster, config=data)
+        cluster.launched_status = "UP"
+
+    @classmethod
+    def check_status(cls, cluster, ping_cluster: bool = False) -> Optional[dict]:
+        """Fetch status for a cluster via Den."""
+        cluster_uri = rns_client.format_rns_address(cluster.rns_address)
+        url = f"{rns_client.api_server_url}/cluster/{cluster_uri}/status?ping={ping_cluster}"
```

Comment:
i dont think we have ping confgiured as a query param in den yet, also pretty sure the default den behavior right now is to send a ping - is the concern here if we do that it will be a bit slow?